src/OrchardCore/Orchard.Environment.Shell/ShellSettingsConfigurationProvider.cs

```diff
@@ -50,23 +50,14 @@ public void SaveToSource(string name, IDictionary<string, string> configuration)
                 Optional = false
             });
 
-            configurationProvider.Set(name, null);
-            configurationProvider.Set($"{name}:RequestUrlHost", ObtainValue(configuration,$"{name}:RequestUrlHost"));
-            configurationProvider.Set($"{name}:RequestUrlPrefix", ObtainValue(configuration,$"{name}:RequestUrlPrefix"));
-            configurationProvider.Set($"{name}:DatabaseProvider", ObtainValue(configuration,$"{name}:DatabaseProvider"));
-            configurationProvider.Set($"{name}:TablePrefix", ObtainValue(configuration,$"{name}:TablePrefix"));
-            configurationProvider.Set($"{name}:ConnectionString", ObtainValue(configuration,$"{name}:ConnectionString"));
-            configurationProvider.Set($"{name}:State", ObtainValue(configuration,$"{name}:State"));
+            foreach (var item in configuration)
+            {
+                configurationProvider.Set($"{item.Key}", item.Value ?? "~");
```

Comment:
We cant do this..... If we have two providers, lets say. One for the Default, and one for Azure or AWS KMS, when we have all keys from these two loaded, if we then save the core Settings back to default, it will flush everything.